### PR TITLE
Fix deploy and update helm/kubectl

### PR DESCRIPTION
**Pull request by consideRatio**
@lwasser I think this may solve the issues you face, which I'm quite confident was caused by a purely numerical git hash being set in a yaml file, that suddenly interpreted the value as an integer rather than a string.

### CODE REVIEW

**Comment by lwasser** [comment]
@consideRatio wow thank you. I just tried this locally and it seemed to deploy from what i can tell! 
I wouldn't not have considered that issue of the commit hash being purely numeric. 

i'm going to check my hub now. THANK YOU. i'm taking a lot of notes as i've learned a lot! This was actually helpful for me to better understand how this all works.

**Comment by lwasser** [comment]
ok one more question. when i try to launch the hub i'm getting that invalid image name error @consideRatio  i tried it for myself and one of my student's pods (someone who doesn't use our hub). 

```
ea-hub        jupyter-calekochenour                                       0/1     InvalidImageName   0          16s
ea-hub        jupyter-lwasser                                             0/1     InvalidImageName   0          3m31s
```

any thoughts? it deployed but it seems like there is still a container issue

**Comment by lwasser** [comment]
a bit more information
```
2020-04-24 15:15:56+00:00 [Warning] Failed to apply default image tag "earthlabhubops/ea-k8s-user-ea-hub:\"9152039\"": couldn't parse image reference "earthlabhubops/ea-k8s-user-ea-hub:\"9152039\"": invalid reference format
Event log
```
it escaped the tag name quotes?

**Comment by consideRatio** [comment]
Awww okay, nice debugging.

Hmmm... deploy.py will run `helm upgrade .... --set jupyterhub.hub.image.tag=...` and we need to make sure it sets the tag to a string, rather than an integer, and I'm not sure how to make it differentiate that when it passes through a shell terminal command like that.

**Comment by consideRatio** [comment]
https://github.com/helm/helm/issues/3155

The solution is to use `--set-string` instead of `--set`!

**Comment by consideRatio** [comment]
> it escaped the tag name quotes?

I think the tag suddenly became parsed as a string, but a string that was `"1234567"` rather than the string `1234567` as we wanted when working to avoid the integer `1234567`.

**Comment by lwasser** [comment]
@consideRatio WOW - that worked perfectly! So take aways here
1. There was an issue with a numeric hash being parsed as a number rather than a string identifier for my container.
2. To fix this, we simply use `--set-string` to force it to read the hash as a string regardless!!

i just pushed locally and my hub updated AND i was able to login successfully. 

Wow thank you again. i'll edit this pr with the set-string call and will merge it. I'm super appreciative of your help with this!!

File: deploy.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def deploy(chartname):
 
         print("Using", image_spec, "as hub image for", chartname)
         extra_args.extend(['--set',
```
Comment (lwasser) [inline deploy.py]:
```suggestion
        extra_args.extend(['--set-string',
```

**Comment by lwasser** [comment]
you're speedy. thank you. this fix works. i'll merge this now. I'm so so so appreciative of this help, erik! thank you AGAIN!!!

**Comment by consideRatio** [comment]
@lwasser wieee =) glad it worked out Leah! Hope I didn't break something by bumping to the latest kubernetes and helm version, I think it is a safe move that can help us avoid some issues down the line, but I'm not 100%.

**Comment by lwasser** [comment]
sure thing!! I see how you updated things and it seems ok right now. I will monitor it as well just in case :) i can always roll that back ! 
 YAY!! thank you again. I just don't know how i would have caught that issue!!  But it's good for me to now know how to check in on cluster!!

**Comment by consideRatio** [comment]
;D @lwasser I opened #200 to fix a mistake I made.

**Comment by lwasser** [comment]
ok great - thank you!! ill have a look now!!

**Pull request merged**